File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/response/analytic/report/field/CoreAnalyticReportFieldDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.response.analytic.report.field;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.infra.reactive.form.engine.form.engine.model.dto.response.element.CoreAllElementDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.table.column.CoreTableColumnDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.window.tab.field.CoreWindowTabFieldDTO;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CoreAnalyticReportFieldDTO {
    private long id;
    private String title;
    private CoreTableColumnDTO coreTableColumnDTO;
    private CoreWindowTabFieldDTO coreWindowTabFieldDTO;
    private List<CoreAnalyticReportFieldAggregateModelDTO> coreAnalyticReportFieldAggregateModelDTOS;
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Response/Analytic/Report/Field/CoreAnalyticReportFieldDTO.js
import {BaseModel} from "../../../../../../../UIFrameWork/Shared/Common/BaseModel.js";
import ConvertUtil from "../../../../../Common/ConvertUtil.js";
import CoreTableColumnDTO from "../../../Table/Column/CoreTableColumnDTO.js";
import CoreWindowTabFieldDTO from "../../../Window/Tab/Field/CoreWindowTabFieldDTO.js";
import CoreAnalyticReportFieldAggregateModelDTO from "./CoreAnalyticReportFieldAggregateModelDTO.js";

export default class CoreAnalyticReportFieldDTO extends BaseModel {
    constructor() {
        super();
    }

    getTitle() {
        return this.title;
    }

    getCoreTableColumnDTO() {
        if (this.coreTableColumnDTO && this.coreTableColumnDTOConverted == null) {
            this.coreTableColumnDTOConverted = ConvertUtil.ConvertGeneral(CoreTableColumnDTO, this.coreTableColumnDTO);
        }
        return this.coreTableColumnDTOConverted;
    }

    getCoreWindowTabFieldDTO() {
        if (this.coreWindowTabFieldDTO && this.coreWindowTabFieldDTOConverted == null) {
            this.coreWindowTabFieldDTOConverted = ConvertUtil.ConvertGeneral(CoreWindowTabFieldDTO, this.coreWindowTabFieldDTO);
        }
        return this.coreWindowTabFieldDTOConverted;
    }

    getCoreAnalyticReportFieldAggregateModelDTOS() {
        if (this.coreAnalyticReportFieldAggregateModelDTOS && this.coreAnalyticReportFieldAggregateModelDTOSConverted == null) {
            this.coreAnalyticReportFieldAggregateModelDTOSConverted = ConvertUtil.ConvertGeneralWithArray(CoreAnalyticReportFieldAggregateModelDTO, this.coreAnalyticReportFieldAggregateModelDTOS);
        }
        return this.coreAnalyticReportFieldAggregateModelDTOSConverted;
    }
}